controllers/machine_controller_phases.go

```diff
@@ -197,6 +197,14 @@ func (r *MachineReconciler) reconcileInfrastructure(ctx context.Context, m *clus
 	if infraConfig == nil && err == nil {
 		return nil
 	} else if err != nil {
+		if m.Status.InfrastructureReady {
```

Comment:
This should probably also test `apierrors.IsNotFound(errors.Cause(err))`, to ensure that the error is related to the resource being missing rather than another type of error.

Reply:
I tried that. It seems reconcileExternal is consuming that error and only returning the requeue error

Reply:
Arrgh, indeed it is...  I'm wondering if we need to refactor `reconcileExternal` a bit so that we are not returning a wrapped RequeueAfter error there....  It would probably require registering the watcher for the resource prior to actually retrieving one, though...  @vincepri @ncdc thoughts?

Reply:
We could either move the requeue error up to the calling functions, or (less preferred, but quicker) string match "not found" on the incoming error

Reply:
@vincepri @detiber I chose the quicker but not preferred way ... the placement of the requeue error in calling function is bit of a design issue since it also requires change in ```reconcileBootstrap``` , which I thought is bit out of scope of this bug, shall we follow this in another PR?

Reply:
I'm good with this as a short term solution as long as a tracking issue is created to clean up later.